### PR TITLE
Add first few Architecture Decision Records (ADRs) for chosen tooling

### PR DESCRIPTION
**Pull request by Gweaton**
## Changes in this PR

Adds the first bunch of ADRs (Architecture Decision Records) to the project to document the motivation behind the architectural decisions we've made on the project, which is helpful for future readers to the codebase.

See: [Documenting architecture decisions](https://cognitect.com/blog/2011/11/15/documenting-architecture-decisions)
for more information.

This adds ADRs for:

- NestJS
- TypeScript
- ESLint
- Cypress
- TypeORM
- Keeping a changelog
- Prettier

### CODE REVIEW

File: doc/architecture/decisions/0002-use-nestjs-as-our-application-framework.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+transfer.
+
+Although there are multiple application frameworks we could have chosen, we
+believe that NestJS provides a well-defined application structure for our needs
+and a consistent and well-documented developer experience. NestJS has a healthy
+established community, good documentation, and is open source.
+
+## Consequences
+
+Choosing a widely-adopted, opinionated framework like Nest JS that wraps around
```
Comment (bryangaledxw) [inline doc/architecture/decisions/0002-use-nestjs-as-our-application-framework.md:39]: "Nest JS" -> "NestJS" for consistency

**Review by bryangaledxw — COMMENTED** [review]
Looks good to me, just that one correction

File: doc/architecture/decisions/0002-use-nestjs-as-our-application-framework.md (hunk 1 above)
Comment (Gweaton) [reply doc/architecture/decisions/0002-use-nestjs-as-our-application-framework.md]: Well-spotted, thanks!

**Review by bryangaledxw — APPROVED** [review]

**Review by pezholio — APPROVED** [review]
Incredible work - thanks for this! 👍

**Pull request merged**